File: backend/src/Api/Profiles/ProfileModels.cs
using System.ComponentModel.DataAnnotations;

namespace StudyOrganizer.Api.Profiles;

public sealed class UpdateProfileRequest
{
    [StringLength(100)]
    public string? FirstName { get; init; }

    [StringLength(100)]
    public string? LastName { get; init; }

    public DateOnly? DateOfBirth { get; init; }

    public string? Gender { get; init; }
}

public sealed record ProfileResponse(
    Guid UserId,
    string Email,
    string? FirstName,
    string? LastName,
    DateOnly? DateOfBirth,
    string? Gender);

File: frontend/src/features/profile/profileModels.ts
export type ProfileGender =
  | 'Female'
  | 'Male'
  | 'PreferNotToSay'

export interface UserProfile {
  userId: string
  email: string
  firstName: string | null
  lastName: string | null
  dateOfBirth: string | null
  gender: ProfileGender | null
}

export interface UpdateProfileRequest {
  firstName: string | null
  lastName: string | null
  dateOfBirth: string | null
  gender: ProfileGender | null
}
